src/pyhf/optimize/opt_minuit.py

```diff
@@ -1,87 +1,115 @@
+            for negative log-likelihood function. See page 37 of
+            http://hep.fi.infn.it/minuit.pdf. This parameter is sometimes
+            called UP in the MINUIT docs.
+
+
         Args:
-            verbose (`bool`): print verbose output during minimization
-        
+            errordef (`float`): See minuit docs. Default is 1.0.
+            steps (`int`): Number of steps for the bounds. Default is 1000.
         """
-        self.verbose = verbose
-        self.ncall = ncall
-        self.errordef = errordef
-        self.steps = steps
+        self.errordef = kwargs.pop('errordef', 1)
+        self.steps = kwargs.pop('steps', 1000)
+        self.name = 'minuit'
+        super(minuit_optimizer, self).__init__(*args, **kwargs)
 
-    def _make_minuit(
-        self, objective, data, pdf, init_pars, init_bounds, fixed_vals=None
+    def _minimize(
+        self,
+        objective,
+        init,
+        method='SLSQP',
+        jac=None,
+        bounds=None,
+        fixed_vals=None,
+        return_uncertainties=False,
+        options={},
     ):
-        def f(pars):
-            result = objective(pars, data, pdf)
-            logpdf = result[0]
-            return logpdf
-
-        parnames = ['p{}'.format(i) for i in range(len(init_pars))]
-        kw = {'limit_p{}'.format(i): b for i, b in enumerate(init_bounds)}
-        initvals = {'p{}'.format(i): v for i, v in enumerate(init_pars)}
+        """
+        Same signature as scipy.optimize.minimize.
+
+        This returns a callable that returns the fitresult.
+
+        Note: an additional `minuit` is injected into the fitresult to get the
+        underlying minimizer.
+
+        Minimizer Options:
+            return_uncertainties (`bool`): Return uncertainties on the fitted parameters. Default is off.
+
+        Returns:
+            fitresult (`scipy.optimize.OptimizeResult`): the fit result
+        """
+        parnames = ['p{}'.format(i) for i in range(len(init))]
+        kw = {'limit_p{}'.format(i): b for i, b in enumerate(bounds)}
+        initvals = {'p{}'.format(i): v for i, v in enumerate(init)}
         step_sizes = {
             'error_p{}'.format(i): (b[1] - b[0]) / float(self.steps)
-            for i, b in enumerate(init_bounds)
+            for i, b in enumerate(bounds)
         }
         fixed_vals = fixed_vals or []
         constraints = {}
         for index, value in fixed_vals:
-            constraints = {'fix_p{}'.format(index): True}
+            constraints['fix_p{}'.format(index)] = True
             initvals['p{}'.format(index)] = value
-        kwargs = {}
-        for d in [kw, constraints, initvals, step_sizes]:
-            kwargs.update(**d)
-        mm = iminuit.Minuit(
-            f,
+
+        self._minimizer = iminuit.Minuit(
+            objective,
             print_level=1 if self.verbose else 0,
-            errordef=1,
+            errordef=self.errordef,
             use_array_call=True,
```

Comment:
As we saw that in my tests that old versions of `iminuit` don't have this API can we set a lower bound in the `minuit` extra? Or do you want that in a different PR?

Reply:
I can just make a small maintenance PR on this, as it isn't really in the scope of this PR.

Reply:
This is now PR #975